packages/walletconnect-v2/src/index.ts

```diff
@@ -86,27 +102,54 @@ export class WalletConnect extends Connector {
     this.events.emit(URI_AVAILABLE, uri)
   }
 
+  private async initializeProvider(
```

Comment:
I am wondering if this PR could be broken down into a smaller diff - the one that actually fixes the issues in PR title. I see a lot of changes like this one - which seem to be more of "refactoring" that are not related to the described issues, unless I am mistaken?

Would definitely understand the reason behind creating this utility function or reordering private class properties.

Reply:
I am also not aware of recent changes in other files, but one thing to keep in mind is that, in principle, we always tried to keep providers similar as far as top-level methods and private APIs they have. If you would look at other existing providers, you would notice they all have `isomorphicInitialize`, but not necessarily have a lot of private/non-standard methods. Something to keep in mind for future readability/ease of maintenance.

Reply:
Upgrading the package should (according to the wc team) get us the improvements mentioned in the title/description. All changes here are made to accommodate the new requirements from `@walletconnect/ethereum-provider`. i.e., enforcing `ArrayOfOneOrMore<T>` for `ChainProps`

Reply:
@grabbou Are you saying here that you'd rather a class import a utility function from another file rather than define its own private utility method?

Reply:
> l changes here are made to accommodate the new requirements from @walletconnect/ethereum-provider. i.e., enforcing ArrayOfOneOrMore<T> for ChainProps

> @grabbou Are you saying here that you'd rather a class import a utility function from another file rather than define its own private utility method?

What are the utility functions that are relevant to the PR apart from the array check? I was referring to the private methods defined on the class, that, to me, look like just extracted from their original places and still being referred only in those places.

I would like to get a full picture - why those changes were necessary and how they relate to the PR